docs/05-concepts/10-authentication/05-custom-integration.md

```diff
@@ -0,0 +1,194 @@
+# Custom integration
+
+Serverpod is designed to make it as simple as possible to implement custom authentication integrations. The framework comes with an integrated auth token creation, validation, and communication system. With a simple setup, it is easy to generate tokens and include them in authenticated communication with the server.
+
+## Server setup
+
+After successfully authenticating a user, for example, through a username and password, an auth token can be created to preserve the authenticated user's permissions. This token is used to identify the user and facilitate endpoint authorization validation. When the user signs out, the token can be removed to prevent further access.
+
+### Create auth token
+
+To create an auth token, call the `signInUser` method in the `UserAuthentication` class, accessible through the `session.auth` field on the `session` object. 
+
+The `signInUser` method takes three arguments: the first is a unique `integer` identifier for the user, the second is information about the method used to authenticate the user, and the third is a set of scopes granted to the auth token.
+
+```dart
+var authToken = await session.auth.signInUser(myUserObject.id, 'myAuthMethod', scopes: {
+    Scope('delete'),
+    Scope('create'),
+});
+```
+
+The example above creates an auth token for a user with the unique identifier taken from `myUserObject`. The auth token preserves that it was created using the method `myAuthMethod` and has the scopes `delete` and `create`.
+
+
+:::info
+The unique identifier for the user should uniquely identify the user regardless of authentication method. The information allows authentication tokens associated with the same user to be grouped.
+:::
+
+#### Custom auth tokens
+
+The `UserAuthentication` class simplifies the token management but makes assumptions about what information should be stored in the auth token. If your project has different requirements, managing auth tokens manually with your defined model is possible. Custom auth tokens require that the token validation is overridden and adjusted to the new auth token format, explained in [override token validation](#override-token-validation). 
+
+### Token validation format
+
+The framework requires tokens to be of `String` type, and the default token validation expects the token to be in the format `userId:key`. The `userId` is the unique identifier for the user, and the `key` is a generated auth token key. The `userId` and `key` are then retrieved from the token and validated towards the auth token stored as a result of the call to `session.auth.signInUser(...)`.
+
+```dart
+var authToken = await session.auth.signInUser(....);
+var verifiableToken = '${authToken.userId}:${authToken.key}';
+```
+
+In the above example, the `verifiableToken` is created by concatenating the `userId` and `key` from the `authToken`. This token is then verifiable by the default token validation.
+
+#### Override token validation
+
```

Comment:
Maybe clarify that this is typically not needed for building your own authentication method.